bigbluebutton-html5/imports/ui/components/audio/audio-settings/component.jsx

```diff
@@ -1,73 +1,101 @@
 import React from 'react';
-import { defineMessages, injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
 import Button from '/imports/ui/components/button/component';
 import { withModalMounter } from '/imports/ui/components/modal/service';
-import styles from '../audio-modal/styles.scss';
-
 import DeviceSelector from '/imports/ui/components/audio/device-selector/component';
-import AudioStreamVolume from '/imports/ui/components/audio/audio-stream-volume/component';
-import EnterAudioContainer from '/imports/ui/components/audio/enter-audio/container';
 import AudioTestContainer from '/imports/ui/components/audio/audio-test/container';
 import cx from 'classnames';
+import styles from './styles';
+
+const propTypes = {
+  intl: intlShape.isRequired,
+  exitAudio: PropTypes.func.isRequired,
+  changeInputDevice: PropTypes.func.isRequired,
+  changeOutputDevice: PropTypes.func.isRequired,
+  handleBack: PropTypes.func.isRequired,
+  handleRetry: PropTypes.func.isRequired,
+  isConnecting: PropTypes.bool.isRequired,
+  inputDeviceId: PropTypes.string.isRequired,
+  outputDeviceId: PropTypes.string.isRequired,
+};
+
+const intlMessages = defineMessages({
+  backLabel: {
+    id: 'app.audio.backLabel',
+    description: 'audio settings back button label',
+  },
+  descriptionLabel: {
+    id: 'app.audio.audioSettings.descriptionLabel',
+    description: 'audio settings description label',
+  },
+  micSourceLabel: {
+    id: 'app.audio.audioSettings.microphoneSourceLabel',
+    description: 'Label for mic source',
+  },
+  speakerSourceLabel: {
+    id: 'app.audio.audioSettings.speakerSourceLabel',
+    description: 'Label for speaker source',
+  },
+  streamVolumeLabel: {
+    id: 'app.audio.audioSettings.microphoneStreamLabel',
+    description: 'Label for stream volume',
+  },
+  retryLabel: {
+    id: 'app.audio.audioSettings.retryLabel',
+    description: 'Retry button label',
+  },
+});
 
 class AudioSettings extends React.Component {
   constructor(props) {
     super(props);
 
-    this.chooseAudio = this.chooseAudio.bind(this);
+    const {
+      handleBack,
+      handleRetry,
+      exitAudio,
+      changeInputDevice,
+      changeOutputDevice,
+    } = props;
+
     this.handleInputChange = this.handleInputChange.bind(this);
     this.handleOutputChange = this.handleOutputChange.bind(this);
-    this.handleClose = this.handleClose.bind(this);
+    this.handleBack = handleBack;
+    this.handleRetry = handleRetry;
```

Comment:
If you are not binding them... then what's the point of these assignments?
Just curious.